sympy/core/assumptions.py

```diff
@@ -285,7 +285,10 @@ def _ask(fact, obj):
     except KeyError:
         pass
     else:
-        a = evaluate(obj)
+        try:
+            a = evaluate(obj)
+        except ZeroDivisionError:
+            return None
```

Comment:
That's overgeneral, it's catching the exception even for things that are not `asec(0)'.
(I don't know offhand what the proper place to fix this would be, maybe somebody else does.)